bmcremedy_connector.py

```diff
@@ -131,21 +157,61 @@ def initialize(self):
+            self.debug_print("Resetting the state file with the default format")
+            self._state = {"app_version": self.get_app_json().get("app_version")}
+            if self.auth_type == consts.BMCREMEDY_OAUTH:
+                return self.set_status(phantom.APP_ERROR, consts.BMCREMEDY_STATE_FILE_CORRUPT_ERROR)
+
+        if len(config.get("ports").split(',')) != 2:
+            return self.set_status(phantom.APP_ERROR, consts.BMCREMEDY_REQUIRED_PARAM_PORTS.format("ports"))
+
+        if self.auth_type == consts.BMCREMEDY_BASIC:
+            required_params = ["username", "password"]
+            for key in required_params:
+                if not config.get(key):
+                    return self.set_status(phantom.APP_ERROR, consts.BMCREMEDY_REQUIRED_PARAM_BASIC.format(key))
+        elif self.auth_type == consts.BMCREMEDY_OAUTH:
+            required_params = [consts.BMCREMEDY_CONFIG_CLIENT_ID, consts.BMCREMEDY_CONFIG_CLIENT_SECRET]
+            for key in required_params:
+                if not config.get(key):
+                    return self.set_status(phantom.APP_ERROR, consts.BMCREMEDY_REQUIRED_PARAM_OAUTH.format(key))
+            if self.get_action_identifier() == phantom.ACTION_ID_TEST_ASSET_CONNECTIVITY:
+                test_asset_connectivity_oauth = True
+            else:
+                if not self._state.get("oauth_token", False):
+                    return self.set_status(phantom.APP_ERROR, "Required tokens not found in state file. Please run test connectivity first...")
+        else:
+            return self.set_status(phantom.APP_ERROR, "Please provide a valid authentication mechanism to use")
+
+        if test_asset_connectivity_oauth:
+            self._base_url += f':{config.get("ports").split(",")[1].strip()}'
+        else:
+            self._base_url += f':{config.get("ports").split(",")[0].strip()}'
+
+        self.is_oauth_token_not_exist = self.auth_type == consts.BMCREMEDY_OAUTH and \
+            (not self._state.get("oauth_token", {}).get("access_token", False)) and \
+                (not self._state.get("oauth_token", {}).get("refresh_token", False))
+        self._is_client_id_changed = (self._state.get(consts.BMCREMEDY_CONFIG_CLIENT_ID) and config.get(consts.BMCREMEDY_CONFIG_CLIENT_ID)) and \
+            self._state.get(consts.BMCREMEDY_CONFIG_CLIENT_ID) != config.get(consts.BMCREMEDY_CONFIG_CLIENT_ID)
+
         # Return response_status
         return phantom.APP_SUCCESS
 
-    def _check_login_status(self, action_result, response):
+    def _check_login_status(self, response):
 
         if not hasattr(response, 'headers'):
-            return action_result.set_status(phantom.APP_ERROR, "Response missing headers, cannot determine success")
+            return phantom.APP_ERROR, "Response missing headers, cannot determine success"
+            # return action_result.set_status(phantom.APP_ERROR, "Response missing headers, cannot determine success")
```

Comment:
Probably a leftover. Can this commented line be deleted? Same for commented lines below.

Reply:
Yes, will remove commented code.